python/remote_api/src/lynx/kite.py

```diff
@@ -865,11 +865,30 @@ def from_df(self, df, index=True):
       return self.uploadParquetNow(f.read())
 
   def from_spark(self, df):
-    """Converts a Spark DataFrame to a LynxKite table."""
-    filename = 'DATA$/python-imports/' + datetime.datetime.now().strftime('%Y-%d-%m_%H%M%S.%f')
-    resolved = self.get_prefixed_path(filename).resolved
-    df.write.parquet(resolved)
-    return self.importParquetNow(filename=filename)
+    '''
+    Converts a Spark DataFrame to a LynxKite table.
+    Costs almost nothing if LynxKite was started in the same SparkSession.
+    '''
+    if self._lynxkite:
+      # The DataFrame is in the same JVM! We can use it directly.
+      guid = self._lynxkite.importDataFrame(df)
+      print('guid:', guid)
+      return SingleOutputAtomicBox(
+          self.box_catalog(), self, 'importParquet', inputs={}, parameters={
+              'filename': 'from PySpark',
```

Comment:
I strategically chose this filename so that the box looks like this:

<img width="173" alt="image" src="https://user-images.githubusercontent.com/1268018/212145112-199004cc-6ae8-4110-a7cf-de895c211b28.png">